rasterio/_base.pyx

```diff
@@ -534,39 +528,44 @@ cdef class DatasetReader(object):
         return self._transform
 
     property transform:
-        """Coefficients of the affine transformation that maps col,row
-        pixel coordinates to x,y coordinates in the specified crs. The
-        coefficients of the augmented matrix are shown below.
+        """An instance of ``affine.Affine``, which is a ``namedtuple`` with
+        coefficients in the order ``(a, b, c, d, e, f)``.
+
+        Coefficients of the affine transformation that maps ``col,row``
+        pixel coordinates to ``x,y`` coordinates in the specified crs. The
+        coefficients of the augmented matrix are:
         
           | x |   | a  b  c | | r |
           | y | = | d  e  f | | c |
           | 1 |   | 0  0  1 | | 1 |
-        
-        In Rasterio versions before 1.0 the value of this property
-        is a list of coefficients ``[c, a, b, f, d, e]``. This form
-        is *deprecated* beginning in 0.9 and in version 1.0 this 
-        property will be replaced by an instance of ``affine.Affine``,
-        which is a namedtuple with coefficients in the order
-        ``(a, b, c, d, e, f)``.
-
-        Please see https://github.com/mapbox/rasterio/issues/86
-        for more details.
         """
         def __get__(self):
-            warnings.warn(
-                    "The value of this property will change in version 1.0. "
-                    "Please see https://github.com/mapbox/rasterio/issues/86 "
-                    "for details.",
-                    FutureWarning,
-                    stacklevel=2)
-            return self.get_transform()
+            return Affine.from_gdal(*self.get_transform())
 
     property affine:
-        """An instance of ``affine.Affine``. This property is a
+        """This property is deprecated.
+
+        An instance of ``affine.Affine``. This property is a
         transitional feature: see the docstring of ``transform``
         (above) for more details.
+
+        This property was added in ``0.9`` as a transitional feature to aid the
+        transition of the `transform` parameter.  Rasterio ``1.0`` completes
+        this transition by converting `transform` to an instance of
+        ``affine.Affine()``.
+
+        See the `transform`'s docstring for more information.
+
+        See https://github.com/mapbox/rasterio/issues/86 for more details.
         """
+
         def __get__(self):
+            warnings.warn(
+                "'src.affine' is deprecated.  Please switch to "
+                "'src.transform'. See "
+                "https://github.com/mapbox/rasterio/issues/86 for details.",
+                RuntimeWarning,
```

Comment:
@perrygeo I think we can do:

```python
with warnings.catch_warnings():
    warnings.simplefilter('always')
    warnings.warn('message', DeprecationWarning)
```

which exposes the `DeprecationWarning` and resets the warning filter in case the end user adjusted it.